### ISSUE TITLE
Thumbnail Generation Service

### ISSUE THREAD

**Issue by micahalcorn** [OriginProtocol/origin#2187]
Build a service for generating various sizes of images that are added to listings (and profiles?) so that only the optimally-sized image is loaded on small screens, for example.

**Comment by tomlinton** [comment]
I think this is a duplicate of #947? I think we ended up deciding against this in the discussion there.

**Comment by micahalcorn** [comment]
Oh, okay, thanks. I misunderstood. 🙂

**Comment by tomlinton** [comment]
I don't think you did, it sounded like we wanted to revisit this from the conversation. I mostly wanted to point out the old discussion. :)

Lets leave this one open and close the other one.

**Comment by nick** [comment]
Could be worth considering this as a quick fix: 

https://blog.cloudflare.com/announcing-cloudflare-image-resizing-simplifying-optimal-image-delivery/

**Comment by nick** [comment]
I have just experimented with using https://github.com/willnorris/imageproxy which looks like a great solution for this due to its speed and caching flexibility. Took me 5 mins to set up locally and reduced image bandwidth on page 1 of shoporigin.com from 3.6Mb to 460k.

I think we can setup images.originprotocol.com and let the DApp use that to auto-resize and cache images. We could also put a CDN in front of that to further reduce image latency.

cc @franckc @mikeshultz

**Comment by franckc** [comment]
Cool. Yes, that seems like a nice image resizer http proxy.
I'm sure it would help on low bandwidth connections. Like on mobile.

Though I bet currently our primary DApp performance bottleneck is not pulling images but rather the latency of our graphQL server...

**Comment by nick** [comment]
Yea agree GraphQL server is the biggest cause of latency. The main reason I looked at this again was to help fix #3570 but I think anything that can help reduce bandwidth would be helpful too.
